libraries/AP_HAL_PX4/UARTDriver.cpp

```diff
@@ -72,19 +71,14 @@ void PX4UARTDriver::begin(uint32_t b, uint16_t rxS, uint16_t txS)
       thrashing of the heap once we are up. The ttyACM0 driver may not
       connect for some time after boot
      */
-	if (rxS != 0 && rxS != _readbuf_size) {
+    if (rxS != 0 && rxS != _readbuf.get_size()) {
```

Comment:
Need to take care of the _initialized flag that's a bit different than other places